src/NuGet.Clients/NuGet.SolutionRestoreManager/RestoreManagerPackage.cs

```diff
@@ -114,5 +137,50 @@ private bool ShouldRestoreOnBuild
                 return packageRestoreConsent.IsAutomatic;
             }
         }
+
+        #region IVsUpdateSolutionEvents4
+
+        public void UpdateSolution_QueryDelayFirstUpdateAction(out int pfDelay)
+        {
+            // check if NuGet lock is already acquired by some other NuGet operation
+            if (LockService.IsLockHeld)
+            {
+                // delay build by setting pfDelay to non-zero
+                pfDelay = 1;
+            }
+            else
+            {
+                var isLockAcquired = LockService.TryAcquireLock(NuGetLockTimeout, out _nuGetLock);
```

Comment:
I'm convinced the value should be checked for null even with apparently redundant first `if`. Let's hear some thoughts from other reviewers.